dsref/load.go

```diff
@@ -10,14 +10,15 @@ import (
 // ErrNoHistory indicates a resolved reference has no HEAD path
 var ErrNoHistory = fmt.Errorf("no history")
 
-// Loader loads and opens a dataset. The only useful implementation of the
-// loader interface is in github.com/qri-io/qri/lib.
-// TODO(b5) - This interface is a work-in-progress
+// ErrRefNotResolved is the error when LoadResolved is called on a non-resolved ref
+var ErrRefNotResolved = fmt.Errorf("ref has not been resolved, cannot load")
+
+// Loader loads datasets
 type Loader interface {
```

Comment:
Looking through the code, I don't think both of these methods shouldn't be on the same interface. I can't find an example looking through the this PR that accepts a `Loader` and uses both methods. (If that exists, then we should keep them combined).

I'd prefer to see:
```go
type Loader interface {
  LoadDataset(ctx context.Context, refstr string) (*dataset.Dataset, error)
}

type ResolvedLoader interface {
  LoadResolved(ctx context.Context, ref Ref, location string) (*dataset.Dataset, error)
}
```

To me this is important to keep the inversion-of-control pattern where subsystems accept a `Loader`. We shouldn't be giving subsystems an escape hatch where they can start dictating load locations

Reply:
I'd rather not introduce additional interfaces where they're not needed. My primary concern is the API surface of `scope` and how many similar but slightly different methods it provides. But, investigating more deeply, it looks like the second method `LoadResolved` can actually be completely removed, just by using `LoadDataset` in a few additional places. Currently only `dataset.get`, `fsi.checkout` and `changes.changereports` use it (ignoring tests), and they can all switch to the new `LoadDataset`. So that kinda sidesteps the issue.

> We shouldn't be giving subsystems an escape hatch where they can start dictating load locations

Very much agree that subsystems should not be aware of the load location.

Reply:
Love it. The `Loader` interface  abstracts a common composition (Parse,Resolve,Load), dropping `LoadResolved` is a win. Double points if the code that replaces it presents a technique for "loading without a loader"